Don't try to use 32BLIT_DIR until we've found it

ali1234 opened the pull request:
This variable is set by find_package after it has found the SDK so
attempting to use it before then won't work in all cases.

ali1234 commented:
This should probably be done in a toolchain file anyway.

ali1234 commented:
According to the comments in the pico import scripts, they should be included before calling `project()`. However, `find_package()` cannot be used before `project()`. This means we cannot use it to find the path to the script. What if instead, we just put:

```
if(PICO_BOARD)
    include(${CMAKE_CURRENT_LIST_DIR}/pico_sdk_import.cmake)
    include(${CMAKE_CURRENT_LIST_DIR}/pico_extras_import.cmake)
endif()
```

into a toolchain file? eg `pico.toolchain` in the root of the sdk, along side the 32blit one.

ali1234 commented:
When I included the helpers after project() I still got a .blit file from the build, but I am unable to test it.

Gadgetoid commented:
Near as I can tell, including this stuff in a `.toolchain` file causes more problems.

I'm using the CMakeLists.txt as-is, but having to specify the full `32BLIT_DIR` path up front, like so:

```
cmake .. -DPICO_SDK_PATH=`pwd`/../../pico-sdk -D32BLIT_DIR=`pwd`/../../32blit-beta  -DPICO_BOARD=pimoroni_picosystem
```

Which is, of course, fairly ugly.

This is one heck of a catch-22 :grimacing:

In toolchain for me:

```
make .. -DPICO_SDK_PATH=`pwd`/../../pico-sdk -D32BLIT_DIR=`pwd`/../../32blit-beta  -DPICO_BOARD=pimoroni_picosystem -DCMAKE_TOOLCHAIN_FILE=`pwd`/../../32blit-beta/pico.toolchain
PICO_SDK_PATH is /home/phil/Development/pico/pico-sdk
PICO platform is rp2040.
CMake Error at /home/phil/Development/pico/pico-extras/CMakeLists.txt:12 (project):
  Language 'C' is currently being enabled.  Recursive call not allowed.


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
```

ali1234 commented:
Hence my comment the other day.

I asked on the forums: https://www.raspberrypi.org/forums/viewtopic.php?f=145&t=319595

For now probably the easiest way to make this a non-problem is to check PICO_BOARD in the boilerplate, rather than in the external cmake that we might not even be able to find.

ali1234 committed:
Don't try to include the pico sdk wrapper unless we are building for pico

Now 32BLIT_DIR only needs to be set for Pico builds... and emscripten builds.
It once again won't be necessary for desktop debug builds.

ali1234 commented:
With the latest 32blit SDK it is no longer necessary to include pico-specific stuff in the cmake at all, so closing this.
